Tests.Unit.Presentation.Web/Services/ReferenceServiceTest.cs

```diff
@@ -423,14 +446,132 @@ public void AddReference_Does_Not_Save_Root_If_Add_Fails()
         }
 
+
+        [Fact]
+        public void UpdateReference_Returns_NotFound_If_Root_Is_NotFound()
+        {
+            //Arrange
+            var id = A<int>();
+            var referenceUuid = A<Guid>();
+            var rootType = A<ReferenceRootType>();
+            ExpectGetRootEntityReturns(id, rootType, Maybe<IEntityWithExternalReferences>.None);
+
+            //Act
+            var result = _sut.UpdateReference(id, rootType, referenceUuid, new ExternalReferenceProperties(A<string>(), A<string>(), A<string>(), false));
+
+            //Assert
+            Assert.True(result.Failed);
+            Assert.Equal(new OperationError("Root entity could not be found", OperationFailure.NotFound), result.Error);
+        }
+
+        [Fact]
+        public void UpdateReference_Returns_Forbidden_If_Modification_Of_Root_Is_Denied()
+        {
+            //Arrange
+            var id = A<int>();
+            var rootType = A<ReferenceRootType>();
+            var referenceUuid = A<Guid>();
+            var entity = new Mock<IEntityWithExternalReferences>();
+            ExpectGetRootEntityReturns(id, rootType, Maybe<IEntityWithExternalReferences>.Some(entity.Object));
+            ExpectAllowModifyReturns(entity.Object, false);
+
+            //Act
+            var result = _sut.UpdateReference(id, rootType, referenceUuid, new ExternalReferenceProperties(A<string>(), A<string>(), A<string>(), false));
+
+            //Assert
+            Assert.True(result.Failed);
+            Assert.Equal(new OperationError("Not allowed to modify root entity", OperationFailure.Forbidden), result.Error);
+        }
+
+        [Fact]
+        public void UpdateReference_Returns_NotFound_If_Reference_Is_Not_Found()
+        {
+            //Arrange
+            var id = A<int>();
+            var rootType = A<ReferenceRootType>();
+            var referenceUuid = A<Guid>();
+            var entity = new Mock<IEntityWithExternalReferences>();
+            ExpectGetRootEntityReturns(id, rootType, Maybe<IEntityWithExternalReferences>.Some(entity.Object));
+            ExpectAllowModifyReturns(entity.Object, true);
+            ExpectGetByUuid(referenceUuid, Maybe<ExternalReference>.None);
+
+            //Act
+            var result = _sut.UpdateReference(id, rootType, referenceUuid, new ExternalReferenceProperties(A<string>(), A<string>(), A<string>(), false));
+
+            //Assert
+            Assert.True(result.Failed);
+            Assert.Equal(new OperationError($"Reference with uuid: {referenceUuid} was not found", OperationFailure.NotFound), result.Error);
+        }
+
+        [Fact]
+        public void UpdateReference_Saves_Root_If_Update_Succeeds()
+        {
+            //Arrange
+            var id = A<int>();
+            var rootType = A<ReferenceRootType>();
+            var referenceUuid = A<Guid>();
+            var title = A<string>();
+            var externalReferenceId = A<string>();
```

Comment:
You can actually use the convenience method two wrap all of these individual vars in one

`A<ExternalReferenceProperties>()`
Clears up some space
also in the others